safe/impact_function_v4/impact_function.py

```diff
@@ -1337,3 +1356,121 @@ def style(self):
         # Let's style the aggregation and analysis layer.
         simple_polygon_without_brush(self.aggregation_impacted)
         simple_polygon_without_brush(self.analysis_impacted)
+
+    @property
+    def provenance(self):
+        """Helper method to gather provenance for exposure_impacted layer.
+
+        :returns: Dictionary that contains all provenance.
+        :rtype: dict
+        """
+        # Environment
+        self._provenance['host_name'] = gethostname()
```

Comment:
After checking that the provenance will be return on `self._is_ready`, I got this problem. The impact function will be set to not ready when it runs the `run` method since there are some exposure, hazard, or aggregation assignment in this method. Is it intended?